cmd/osm-bootstrap/crds/config_meshconfig.yaml

```diff
@@ -31,9 +31,253 @@ spec:
+                  type: object
+                  required:
+                    - serviceCertValidityDuration
+                    - certKeyBitSize
+                  properties:
+                    serviceCertValidityDuration:
+                      description: Sets the service certificate validity duration, represented as a sequence of decimal numbers each with optional fraction and a unit suffix.
+                      type: string
+                    certKeyBitSize:
+                      description: Sets the certificate key bit size for data plane certificates.
+                      type: integer
+                    ingressGateway:
+                      description: Configuration for the ingress gateway's certificate
+                      type: object
+                      required:
+                        - subjectAltNames
+                        - validityDuration
+                        - secret
+                      properties:
+                        subjectAltNames:
+                          description: Subject Alternative Names secured by the certificate
+                          type: array
+                          items:
+                            type: string
+                          minItems: 1
+                        validityDuration:
+                          description: Certificate validity duration, represented as a sequence of decimal numbers each with optional fraction and a unit suffix
+                          type: string
+                        secret:
+                          description: Secret reference to store the certificate in
+                          type: object
+                          required:
+                            - name
+                            - namespace
+                          properties:
+                            name:
+                              description: Name of the secret
+                              type: string
+                            namespace:
+                              description: Namespace of the secret
+                              type: string
+                featureFlags:
+                  description: OSM feature flags
+                  type: object
+                  properties:
+                    enableWASMStats:
+                      type: boolean
+                    enableEgressPolicy:
+                      type: boolean
+                    enableMulticlusterMode:
+                      type: boolean
+                    enableSnapshotCacheMode:
+                      type: boolean
+                    enableAsyncProxyServiceMapping:
+                      type: boolean
+                    enableIngressBackendPolicy:
+                      type: boolean
+                    enableEnvoyActiveHealthChecks:
+                      type: boolean
+                    enableRetryPolicy:
+                      type: boolean
+    - name: v1alpha2
+      served: true
+      storage: false
+      deprecated: true
```

Comment:
I forgot to point this out in my review, but we should only mark a field as deprecated once the control plane is updated to use the new version.

Reply:
Ah makes sense. I'll keep that in mind for next time; this PR was done all out of order, so the next feature should be a lot more organized.